src/cleanup_rules/java_dyn/rules.toml

```diff
@@ -321,49 +249,56 @@ is_seed_rule = false
 #
 [[rules]]
 groups = ["boolean_expression_simplify"]
-name = "simplify_equals_equals_true"
-query = """
-(
-    (binary_expression
-        left : [(parenthesized_expression (_) @lhs) (_) @lhs]
-        operator:"=="
-        right: [(parenthesized_expression (_) @rhs) (_) @rhs]
-    )
-@binary_expression
-(#eq? @lhs @rhs)
-)"""
+name = "simplify_equals_equals_true_1"
+query = """dyn :[lhs] == :[lhs]"""
 replace = "true"
 replace_node = "binary_expression"
 is_seed_rule = false
 
+[[rules]]
+groups = ["boolean_expression_simplify"]
+name = "simplify_equals_equals_true_2"
+query = """dyn (:[lhs]) == :[lhs]"""
+replace = "true"
+replace_node = "binary_expression"
+is_seed_rule = false
 
-# Before: 
-#  false == true 
+
+[[rules]]
+groups = ["boolean_expression_simplify"]
+name = "simplify_equals_equals_true_3"
+query = """dyn :[lhs] == (:[lhs])"""
+replace = "true"
+replace_node = "binary_expression"
+is_seed_rule = false
+
+
+# Before:
+#  false == true
 #  true == false
 # After :
 #  false
 #  false
 #
 [[rules]]
 groups = ["boolean_expression_simplify"]
-name = "simplify_equals_equals_false"
-query = """
-(
-    (binary_expression
-        left : [(false) @lhs (true) @lhs (string_literal) (decimal_integer_literal) @lhs (decimal_floating_point_literal) @lhs]
-        operator:"=="
-        right: [(false) @rhs (true) @rhs (string_literal) @rhs (decimal_integer_literal) @rhs (decimal_floating_point_literal) @rhs]
-    )
-@binary_expression
-(#not-eq? @lhs @rhs)
-)"""
+name = "simplify_equals_equals_false_1"
+query = """dyn false == true"""
```

Comment:
I see how a `where` clause could help. 

We could be like `:[x] == :[y] where x != y`